### PR TITLE
Time-boxed entries for the banner

### PR DESCRIPTION
**Pull request by ricardoamaro**
Dynamically manage and display banner entries as suggested in https://github.com/open-telemetry/opentelemetry.io/issues/4380

1. Store the banner details (including end dates) in a the _index.md file. 
1. Use templating in `banners.md` file to sort the entries by the end dates (.to:)
1. Filter out those that have expired

### CODE REVIEW

**Comment by ricardoamaro** [comment]
It should be good. Here is a test:
https://deploy-preview-4429--opentelemetry.netlify.app/
image

File: data/banners.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  message:
+    '[**OTel Community Day** on June
+    25th](https://sessionize.com/OTel-Community-Day/)!'
+  endDate: 2024-06-25
+
+- title: 'OTel and Prometheus Survey'
+  message:
+    'Help improve OTel and Prometheus interoperability: [complete our
+    survey](https://forms.gle/bZAG9f7udoJsjZUG9) by May 31'
+  endDate: 2024-05-31
```
Comment (svrnm) [inline data/banners.yaml:11]:
```suggestion
  endDate: 2024-05-31

- title: Roadmap to v1 for the OpenTelemetry Collector
  message: '[Learn about our roadmap to v1 for the OpenTelemetry Collector](/blog/2024/collector-roadmap/)'
  endDate: 2024-05-21
```

Can you add that for experimentation? If the code works as expecrted it should drop the community day for now

File: data/banners.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+- title: 'OTel Community Day'
+  message:
+    '[**OTel Community Day** on June
+    25th](https://sessionize.com/OTel-Community-Day/)!'
+  endDate: 2024-06-25
```
Comment (svrnm) [inline data/banners.yaml:5]:
```suggestion
  message:
    'Join us for [**OTel Community Day** on June
    25th](https://events.linuxfoundation.org/open-telemetry-community-day/)!'
  endDate: 2024-06-25
```

File: layouts/partials/banner.md [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+  {{/* Limit to the two entries with the closest end dates */}}
+  {{ $entriesToShow := first 2 $sortedAndFiltered }}
 
 <div class="o-banner">
 
-<i class="fas fa-bullhorn"></i> Join us for 
-[**OTel Community Day** on June 25th](https://sessionize.com/OTel-Community-Day/)!
+  {{ range $entriesToShow }}
 
+<i class="fas fa-bullhorn"></i>
```
Comment (svrnm) [inline layouts/partials/banner.md:21]:
```suggestion
<i class="fas fa-bullhorn"></i> 
```

Can we make that bullhorn also configurable? So default to it, if people do not provide another icon from font awesome. We do something similar in the `hugo.yaml` for the icons on the bottom of the page:

```
      - name: Mailing Lists
        url: https://github.com/open-telemetry/community#mailing-lists
        icon: fa fa-envelope
        desc: List of mailing lists that the project uses.
      - name: Mastodon
        url: https://fosstodon.org/@opentelemetry
        icon: fab fa-mastodon
        desc: Follow us on Mastodon to get the latest news!
```

**Review by svrnm — CHANGES_REQUESTED** [review]
Thanks! Looks like it works as expected, a few additional suggestions

File: layouts/partials/banner.md (hunk 3 above)
Comment (ricardoamaro) [reply layouts/partials/banner.md:21]: for sure we can!

**Comment by chalin** [comment]
Thanks @ricardoamaro for your contribution.

All: Setting this to draft until we agree on a way to move forward. For details, see https://github.com/open-telemetry/opentelemetry.io/issues/4380#issuecomment-2097815727.

**Comment by ricardoamaro** [comment]
No problem. This is how it looks right now 
image

**Comment by ricardoamaro** [comment]
@svrnm @chalin changes added
image
I left the example content for review

**Comment by chalin** [comment]
Thanks for the update. I'll take a look as soon as I can. GH says that there are potential conflicts; please rebase and resolve them. Thx

**Review by theletterf — APPROVED** [review]

**Comment by svrnm** [comment]
> Please also let me know if we still should limit to 2 entries.

Yes, we might consider that sometimes we want to have more than 2, but this should be the default, otherwise this is in a really good shape, thx.

**Review by svrnm — APPROVED** [review]

**Review by chalin — CHANGES_REQUESTED** [review]
Thanks for the updates @ricardoamaro. See inline comments.

@svrnm - even after the suggested edits are addressed, this should be put on hold until we get an answer to https://github.com/open-telemetry/opentelemetry.io/issues/2402#issuecomment-2108941668.

**Review by chalin — APPROVED** [review]
Thanks for making the suggested changes. LGTM

**Pull request merged**